src/Object/Log/ParsedLog.php

```diff
@@ -0,0 +1,147 @@
+<?php
+/**
+ * @copyright Copyright (C) 2021, Friendica
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace Friendica\Object\Log;
+
+/**
+ * Parse a log line and offer some utility methods
+ */
+class ParsedLog
+{
+	const REGEXP = '/^(\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}[^ ]*) (\w+) \[(\w*)\]: (.*)/';
+
+	/** @var int */
+	public $id = 0;
+
+	/** @var string */
+	public $date = null;
+
+	/** @var string */
+	public $context = null;
+
+	/** @var string */
+	public $level = null;
+
+	/** @var string */
+	public $message = null;
+
+	/** @var string */
+	public $data = null;
+
+	/** @var string */
+	public $source = null;
+
+	/** @var string */
+	public $logline;
+
+	/**
+	 * @param int line id
+	 * @param string $logline Source log line to parse
+	 */
+	public function __construct(int $id, string $logline)
+	{
+		$this->id = $id;
+		$this->parse($logline);
+	}
+
+	private function parse($logline)
+	{
+		// if data is empty is serialized as '[]'. To ease the parsing
+		// let's replace it with '{""}'. It will be replaced by null later
+		$logline = str_replace(' [] - {', ' {""} - {', $logline);
+
+		// here we hope that there will not be the string ' - {' inside the $jsonsource value
+		list($logline, $jsonsource) = explode(' - {', $logline);
+		$jsonsource                 = '{' . $jsonsource;
+
+		$jsondata = null;
+
+		if (strpos($logline, '{"') > 0) {
+			list($logline, $jsondata) = explode('{"', $logline, 2);
+
+			$jsondata = '{"' . $jsondata;
+		}
+
+		preg_match(self::REGEXP, $logline, $matches);
+
+		$this->date    = $matches[1];
+		$this->context = $matches[2];
+		$this->level   = $matches[3];
+		$this->message = trim($matches[4]);
+		$this->data    = $jsondata == '{""}' ? null : $jsondata;
+		$this->source  = $jsonsource;
+		$this->try_fix_json();
+
+		$this->logline = $logline;
+	}
+
+	/**
+	 * Fix message / data split
+	 *
+	 * In log boundary between message and json data is not specified.
+	 * If message  contains '{' the parser thinks there starts the json data.
+	 * This method try to parse the found json and if it fails, search for next '{'
+	 * in json data and retry
+	 */
+	private function try_fix_json()
```

Comment:
Please use `camelCase` for method names as you did in `ParsedLogIterator`
```suggestion
	private function tryFixJson()
```
Please don't forget to update all the references to this method.